tests/issues/test_655.py

```diff
@@ -0,0 +1,155 @@
+
+    },
+    {
+        "name": "dup at exon/intron boundary",
+        "var_c": "NM_999999.1:c.39+1dup",
+        "var_p": "MOCK:p.?",
+        "exc": HGVSError
+    },
+]
+
+real_cases = [
+    {
+        "name": "ins at intron/exon boundary",
+        "var_c": "NM_004380.2:c.3251-1_3251insA",
+        "var_p": "NP_004371.2:p.(Ile1084AsnfsTer3)"
+    },
+    {
+        "name": "ins at exon/intron boundary",
+        "var_c": "NM_004380.2:c.3250_3250+1insT",
+        "var_p": "NP_004371.2:p.(Phe1085LeufsTer2)"
+    },
+    {
+        "name": "dup at intron/exon boundary",
+        "var_c": "NM_024529.4:c.132-2_132-1dup",
+        "var_p": "NP_078805.3:p.(Thr45GlyfsTer65)",
+        "exc": HGVSError
+    },
+    {
+        "name": "dup at intron/exon boundary",
+        "var_c": "NM_004985.4:c.112-8_112-1dup",
+        "var_p": "NP_004976.2:p.(Asp38LeufsTer10)",
+        "exc": HGVSError
+    },
+    {
+        "name": "dup at intron/exon boundary on negative strand",
+        "var_c": "NM_004380.2:c.3251-1dup",
+        "var_p": "NP_004371.2:p.(Ile1084SerfsTer3)",
+        "exc": HGVSError
+    },
+    {
+        "name": "dup at exon/intron boundary",
+        "var_c": "NM_024529.4:c.131+1_131+3dup",
+        "var_p": "NP_078805.3:p.(Thr45Ter)",
+        "exc": HGVSError
+    },
+    {
+        "name": "dup at exon/intron boundary on negative strand",
+        "var_c": "NM_004985.4:c.111+1_111+4dup",
+        "var_p": "NP_004976.2:p.(Asp38ValfsTer11)",
+        "exc": HGVSError
+    },
+    {
+        "name": "snv with broken cigar mapping",
+        "var_c": "NM_014638.2:c.515-99G>T",
+        "var_p": "NP_055453.2:p.?"
+    },
+    {
+        "name": "ins with broken cigar mapping",
+        "var_c": "NM_004799.2:c.71-7048_71-7047insATAT",
+        "var_p": "NP_004790.2:p.?"
+    },
+    {
+        "name": "del with broken cigar mapping",
+        "var_c": "NM_007324.2:c.71-7826_71-7802del",
+        "var_p": "NP_015563.2:p.?"
+    },
+]
+
+
+@pytest.fixture(scope="module")
+def hp():
+    return hgvs.parser.Parser()
+
+
+@pytest.fixture(scope="module")
+def hdp():
+    return hgvs.dataproviders.uta.connect(
+        mode=os.environ.get("HGVS_CACHE_MODE", "run"), cache=CACHE
+    )
+
+
+@pytest.fixture(scope="module")
+def vm(hdp):
+    return hgvs.variantmapper.VariantMapper(hdp)
+
+
+@pytest.fixture(scope="module")
+def am37(hdp):
+    return hgvs.assemblymapper.AssemblyMapper(hdp, assembly_name="GRCh37")
+
+
+@pytest.fixture(scope="module")
+def mock_hdp():
+    fn = os.path.join(os.path.dirname(__file__), "..", "data", "sanity_cp.tsv")
+    return mock_input_data_source.MockInputSource(fn)
+
+
+@pytest.fixture(scope="module")
+def mock_vm(mock_hdp):
+    return hgvs.variantmapper.VariantMapper(mock_hdp, prevalidation_level="INTRINSIC")
+
+
+@pytest.fixture()
+def ref_at_boundary_is_exonic():
+    original = hgvs.global_config.mapping.ref_at_boundary_is_intronic
+    hgvs.global_config.mapping.ref_at_boundary_is_intronic = False
```

Comment:
If I am reading this correctly, we are running all tests with this fixture as `ref_at_boundary_is_intronic = False`, but the default is `True`. Do we have tests that verify the default behavior as well? Besides that, this all looks good to me. Thank you for the detailed oriented work that is necessary for this PR!

Reply:
In my most recent changes I updated the tests I added in this PR to cover both cases of the new configuration value. In addition, all of the existing tests also cover the case where `ins_at_boundary_is_intronic` is `True` since that's the default value.